atmMachine.playground/Contents.swift

```diff
@@ -0,0 +1,135 @@
+
+
+class Person {
+
+    let name: String
+    var balance: Int {
+        willSet {
+            print("\(name) has \(newValue)")
+        }
+    }
+    
+    init(name: String, balance: Int) {
+        self.name = name
+        self.balance = balance
+    }
+    
+}
+
+protocol Withdraws {
+    func withdraw(amount: Int, from person: Person) -> Bool
+}
+
+class Note: Hashable {
```

Comment:
Consider if this could be a value type and also see if the default conformances of the `Equatable` and `Hashable` protocols are sufficient for your purposes.